src/Components/DataHistoryExplorer/Internal/TimeSeriesViewer/Internal/TimeSeriesChart/Internal/ChartCommandBar/ChartCommandBar.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+    IChartCommandBarProps,
+    IChartCommandBarStyleProps,
+    IChartCommandBarStyles
+} from './ChartCommandBar.types';
+import { getStyles } from './ChartCommandBar.styles';
+import {
+    classNamesFunction,
+    useTheme,
+    styled,
+    CommandBar,
+    ICommandBarItemProps
+} from '@fluentui/react';
+import { IDataHistoryChartOptions } from '../../../../../../../../Models/Types/Generated/3DScenesConfiguration-v1.0.0';
+import { QuickTimeSpanKey } from '../../../../../../../../Models/Constants/Enums';
+import {
+    AggregationTypeDropdownOptions,
+    QuickTimeSpans
+} from '../../../../../../../../Models/Constants/Constants';
+import produce from 'immer';
+import { useTranslation } from 'react-i18next';
+import {
+    getQuickTimeSpanKeyByValue,
+    getQuickTimeSpanOptions,
+    getYAxisTypeOptions
+} from '../../../../../../../../Models/SharedUtils/DataHistoryUtils';
+import { capitalizeFirstLetter } from '../../../../../../../../Models/Services/Utils';
+
+const getClassNames = classNamesFunction<
+    IChartCommandBarStyleProps,
+    IChartCommandBarStyles
+>();
+
+const ChartCommandBar: React.FC<IChartCommandBarProps> = (props) => {
+    const { defaultOptions, onChange, deeplink, styles } = props;
+
+    // state
+    const [chartOptions, setChartOptions] = useState<IDataHistoryChartOptions>(
```

Comment:
Yes the common ancestor (`TimeSeriesChart`) stores the main chartOptions state variable. I needed to set a state variable of chartOptions in the parent component of TimeSeriesChart because I needed to sync the chart options between two sibling children (ChartCommandBar and HighChartsWrapper). This is needed to let the consistent chart option communication between these siblings:
![image](https://user-images.githubusercontent.com/45217314/213814449-89d1cc12-a7b9-4f9a-81c3-bf37515e7960.png)

With the internal state in ChartCommandBar, it is just to make it a self contained component where you can specify the default options, so that it is not purely a rendering component with passed values.

Does it make sense?